torchvision/prototype/datasets/_builtin/hmdb51.py

```diff
@@ -0,0 +1,116 @@
+import functools
+import pathlib
+import re
+from typing import Any, Dict, List, Tuple, BinaryIO
+
+from torchdata.datapipes.iter import IterDataPipe, Mapper, Filter, CSVDictParser, IterKeyZipper
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    HttpResource,
+    OnlineResource,
+)
+from torchvision.prototype.datasets.utils._internal import (
+    INFINITE_BUFFER_SIZE,
+    getitem,
+    path_accessor,
+    hint_sharding,
+    hint_shuffling,
+)
+from torchvision.prototype.features import EncodedVideo, Label
+
+
+class HMDB51(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "hmdb51",
+            homepage="https://serre-lab.clps.brown.edu/resource/hmdb-a-large-human-motion-database/",
+            dependencies=("rarfile",),
+            valid_options=dict(
+                split=("train", "test"),
+                split_number=("1", "2", "3"),
+            ),
+        )
+
+    def _extract_videos_archive(self, path: pathlib.Path) -> pathlib.Path:
+        folder = OnlineResource._extract(path)
+        for rar_file in folder.glob("*.rar"):
+            OnlineResource._extract(rar_file)
+            rar_file.unlink()
+        return folder
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        url_root = "https://serre-lab.clps.brown.edu/wp-content/uploads/2013/10"
+
+        splits = HttpResource(
+            f"{url_root}/test_train_splits.rar",
+            sha256="229c94f845720d01eb3946d39f39292ea962d50a18136484aa47c1eba251d2b7",
+        )
+        videos = HttpResource(
+            f"{url_root}/hmdb51_org.rar",
+            sha256="9e714a0d8b76104d76e932764a7ca636f929fff66279cda3f2e326fa912a328e",
+        )
+        videos._preprocess = self._extract_videos_archive
```

Comment:
@NicolasHug The archive is a rar of rars so using a single `extract=True` won't cut it. We need the full extraction since reading from rar archives is rather slow and with this we get a significant performance increase.

Another option would be to use this "recursive extraction" by default when setting `extract=True`.